bin/docserv

```diff
@@ -141,11 +147,12 @@ class Deliverable:
             cmd = shlex.split(commands[i]['cmd'])
             logger.debug("Thread %i: %s" % (thread_id, commands[i]['cmd']))
             time.sleep(1)
-            #s = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            #s.communicate()[0]
-            #if commands[i]['ret_val'] is not None and not commands[i]['ret_val'] == int(s.returncode):
-            #    logger.warning("Build failed! Unexpected return value %i for '%s'" % (s.returncode, commands[i]['cmd']) )
-            #    return False
+            s = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            s.communicate()[0]
+            if commands[i]['ret_val'] is not None and not commands[i]['ret_val'] == int(s.returncode):
+                logger.warning("Build failed! Unexpected return value %i for '%s'" % (s.returncode, commands[i]['cmd']) )
```

Comment:
Use `,` (comma) instead of `%` for the logger, like this:

```python
logger.warning("Build failed! Unexpected return value %i for '%s'", (s.returncode, commands[i]['cmd']) )
```

The longer story is: the logger creates the string only, if the warning level is reached. If you use  `%`, you circumvent this and create the complete string.

For small objects it isn't a problem. But keep in mind, sometimes you want to log objects that's are difficult to convert it to a string. This can be the logging time consuming, especially if the logger is called a lot of times.